Firestore/core/test/unit/core/pipeline/collection_group_test.cc

```diff
@@ -0,0 +1,387 @@
+/*
```

Comment:
Many of these tests have built in assumption that collectionGroup returns documents ordered by document id. This is NOT a requirement, in fact:

"As with all other stages, the order of results from these input stages is not stable. A sort(...) operator should always be added if a specific ordering is desired." https://docs.google.com/document/d/1kyQzJs1jN0ntWydSK1aUSThUIjtlTiSdU9UtiGV8Nd4/edit?usp=sharing&resourcekey=0-U1zwV8bP87jt93iDgRwgcg

Having to sort results imposes additional computation, and possibly prevents optimization.